aws-api/src/main/java/com/amplifyframework/api/aws/auth/AuthRuleRequestDecorator.java

```diff
@@ -83,13 +83,16 @@ public <R> GraphQLRequest<R> decorate(
         AppSyncGraphQLRequest<R> appSyncRequest = (AppSyncGraphQLRequest<R>) request;
         AuthRule ownerRuleWithReadRestriction = null;
         Map<String, Set<String>> readAuthorizedGroupsMap = new HashMap<>();
+        boolean subscribeAllowedForNonOwner = false;
 
         // Note that we are intentionally supporting only one owner rule with a READ operation at this time.
         // If there is more than one, the operation will fail because AppSync generates a parameter for each
         // one. The question then is which one do we pass. JavaScript currently doesn't support this use case
         // and it's not clear what a good solution would be until AppSync supports real time filters.
         for (AuthRule authRule : appSyncRequest.getModelSchema().getAuthRules()) {
-            if (isReadRestrictingOwner(authRule)) {
+            if (doesRuleAllowNonOwnerSubscribe(authRule, authType)) {
+                subscribeAllowedForNonOwner = true;
```

Comment:
we might be able to `return request` here if the remaining auth rules does not matter, since we're doing `!subscribeAllowedForNonOwner` below

Reply:
We could return but doing a return buried in a loop is not encouraged for clarity purposes. What I can do here is `break` though as there is no reason to continue with the loop.